src/ark/utils/deepcell_service_utils.py

```diff
@@ -11,17 +10,97 @@
 import numpy as np
 import requests
 from alpineer import image_utils, io_utils, load_utils, misc_utils
-from requests.adapters import HTTPAdapter
-from requests.exceptions import RetryError
 from tifffile import imread
 from tqdm.notebook import tqdm
-from urllib3 import Retry
+
+
+def zip_input_files(deepcell_input_dir, fov_group, batch_num):
+    """Helper function which handles zipping the batch fov images into a single zip file.
+
+    Args:
+        deepcell_input_dir (str): path to where deepcell input image files are stored
+        fov_group (list): list of fovs to process in this batch
+        batch_num (int): the batch number
+    Returns:
+        str: path to deepcell input zip file
+    """
+
+    # write all files to the zip file
+    zip_path = os.path.join(deepcell_input_dir, f'fovs_batch_{batch_num}.zip')
+
+    # create zip files, skip any existing
+    if not os.path.exists(zip_path):
+        print('Zipping preprocessed tiff files.')
+        with ZipFile(zip_path, 'w', compression=ZIP_DEFLATED) as zipObj:
+            for fov in fov_group:
+                # file has .tiff extension
+                basename = fov + '.tiff'
+                filename = os.path.join(deepcell_input_dir, basename)
+                zipObj.write(filename, basename)
+    else:
+        print(f'fovs_batch_{batch_num}.zip already exists.')
+
+    return zip_path
+
+
+def extract_deepcell_response(deepcell_output_dir, fov_group, batch_num, wc_suffix, nuc_suffix):
+    """Helper function to extract the segmentation masks from the deepcell output zip file.
+
+    Args:
+        deepcell_output_dir (str):
+            path to where deepcell output zips are stored
+        fov_group (list):
+            list of fovs to process in this batch
+        batch_num (int):
+            the batch number
+        wc_suffix (str):
+            Suffix for whole cell DeepCell output filename. e.g. for fovX, DeepCell output
+            should be `<fovX>+suffix.tif`.
+            Whole cell DeepCell files by default get suffixed with `'feature_0'`,
+            it will be renamed to this arg.
+        nuc_suffix (str):
+            Suffix for nuclear DeepCell output filename. e.g. for fovX, DeepCell output
+            should be `<fovX>+suffix.tif`.
+            Nuclear DeepCell files by default get suffixed with `'feature_1'`,
+            it will be renamed to this arg.
+    """
+
+    # extract the .tif output
+    print("Extracting tif files from DeepCell response.")
```

Comment:
Seems like there's a lot of print statements that will trigger on every batch, and they don't have the batch name in them. Did you run this notebook with 3-4 different batches? Is it just a string of the same print statements over and over again?

Reply:
Yeah each batch gets a zip, upload, and extract print statement (plus a progress bar). I just kept it pretty consistent with the prior notebook output, but we can change it if you think it's a bit excessive. Or we could add batch numbers to the statements.

Previous output:
<img width="792" alt="Screenshot 2023-07-25 at 10 22 33 AM" src="https://github.com/angelolab/ark-analysis/assets/38049893/7d381426-f618-449b-8730-63af3321cfa7">

New output:
<img width="737" alt="Screenshot 2023-07-25 at 10 26 36 AM" src="https://github.com/angelolab/ark-analysis/assets/38049893/0dd85377-8e9c-4e4e-9afb-a93dc8966388">

Reply:
I think it would be better to keep the first one that says # of batches, modify the segmentation_progress one so it says progress for batch x, and remove the rest.